internal/edgeapi/backend/factory/factory.go

```diff
@@ -0,0 +1,44 @@
+package factory
+
+import (
+	"go.uber.org/zap"
+	"k8s.io/client-go/tools/record"
+	kubeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedevice"
+	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedeviceset"
+	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedevicesignedrequest"
+	"github.com/project-flotta/flotta-operator/internal/common/repository/edgeworkload"
+	"github.com/project-flotta/flotta-operator/internal/common/storage"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/backend"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/backend/k8s"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/configmaps"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/devicemetrics"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/images"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/k8sclient"
+)
+
+func CreateBackend(initialDeviceNamespace string, client kubeclient.Client, logger *zap.SugaredLogger, eventRecorder record.EventRecorder) backend.EdgeDeviceBackend {
```

Comment:
if we named the package backendfactory, then the method could have been called simply `Create(...)`.
we don't expect the factory to create additional resources, so we can change it to that.